receiver/haproxyreceiver/README.md

```diff
@@ -0,0 +1,55 @@
+# HAProxy Receiver
+
+| Status                   |               |
+| ------------------------ |---------------|
+| Stability                | [development] |
+| Supported pipeline types | metrics       |
+| Distributions            | [contrib]     |
+
+The HAProxy receiver generates metrics by polling periodically the HAProxy process through a dedicated socket or HTTP URL.
+
+Supported pipeline types: metrics
+
+## Getting Started
+
+## Configuration
+
+### endpoint (required)
+Path to the endpoint exposed by HAProxy for communications. It can be a local file socket or a HTTP URL.
+
+### Collection interval settings (optional)
+The scraping collection interval can be configured.
+
+Default: 1 minute
+
+### Example configuration
+
+```yaml
+haproxy:
+  endpoint: file:/var/run/haproxy.ipc
```

Comment:
This would need to be `file:///` since things like `url.Parse` expect the `://` to be there and the additional `/` is the start of the path.

You may want to borrow the structure from the `chronyreceiver` since I imagine they have a similar setup.